src/main/java/mike/controller/Controller.java

```diff
@@ -7,90 +8,100 @@
 import javax.swing.JPanel;
 
 import mike.datastructures.Model;
+import mike.view.GUIView;
+import mike.view.ViewTemplate;
 import mike.datastructures.Entity;
 
 public class Controller {
 
-	private static Boolean changed = false;
-	private static Path path = null;
-	private static Model model = new Model();
-	private static JLabel inClass = null;
+	private boolean changed = false;
+	private Path path = null;
+	private Model Cmodel;
+	private ViewTemplate view = null;
+	private JLabel inClass = null;
 	
-	public Controller (){
+	public Controller (Model model, ViewTemplate views){
+	    Cmodel = model;
+	    view = views;
+	    SaveController.saveListener((JButton) ((GUIView) views).getMenuBar().getComponent(0), this);
+	    SaveController.saveAsListener((JButton) ((GUIView) views).getMenuBar().getComponent(1), this);
+	    LoadController.loadListener((JButton) ((GUIView) views).getMenuBar().getComponent(2), this);
+	    FrameController.addClassListener((JButton) ((GUIView) views).getMenuBar().getComponent(3), this);
+	    ClassController.editModeListener((JButton) ((GUIView) views).getMenuBar().getComponent(4), this);
+	    FrameController.exitListener(changed, view);
```

Comment:
Related to Stefan's comment:

When this listener gets called, it will be created with the value of changed (always false) rather than exitListener getting the value of changed when it needs to.
So exitListener should taken in the Controller object, and call control.getChanged() kind of like the saveListener runs control.setChanged(false).